High load while IDLE under FreeBSD

On my system, alacritty is causing about 2.5-3.0 CPU load (according to htop) while only showing a prompt. Other terminals, st for example, show a load of 0.0 when similary idle...

Via dtrace(1) i gathered the syscalls done by the system, and this showed me, that alacritty generates 2774 syscalls in 3.4s.

Those are the syscalls generated by alacritty during idle:
```
  compat11.kevent                                                  10
  recvmsg                                                          50
  close                                                           639
  compat11.stat                                                   639
  fstatfs                                                         639
  nanosleep                                                       639
  open                                                            639
  getdirentries                                                  1278
  compat11.lstat                                                 1917
```
truss(1) showed, that those open/getdirentries calls were done on $HOME/.config/alacritty and the config file it is containing.

To me it seems that alacritty is really busy querying its XDG_CONFIG dir and file (at least on FreeBSD).

### System
OS: FreeBSD 12.1p6
Version: alacritty 0.4.3
Linux/BSD: X11, Openbox

It seems not that easy to solve, the build fails:
```
# cargo test
    Updating crates.io index
  Downloaded crossbeam-channel v0.4.2
  Downloaded serde_json v1.0.55
  Downloaded libc v0.2.71
  Downloaded anymap v0.12.1
  Downloaded itoa v0.4.6
  Downloaded serde v1.0.114
  Downloaded crossbeam-utils v0.7.2
  Downloaded ryu v1.0.5
   Compiling autocfg v1.0.0
   Compiling maybe-uninit v2.0.0
   Compiling libc v0.2.71
   Compiling cfg-if v0.1.10
   Compiling lazy_static v1.4.0
   Compiling bitflags v1.2.1
   Compiling ryu v1.0.5
   Compiling serde v1.0.114
   Compiling serde_json v1.0.55
   Compiling same-file v1.0.6
   Compiling anymap v0.12.1
   Compiling itoa v0.4.6
   Compiling walkdir v2.3.1
   Compiling crossbeam-utils v0.7.2
   Compiling filetime v0.2.10
   Compiling crossbeam-channel v0.4.2
   Compiling notify v5.0.0-pre.3 (/usr/home/manuel/devel/github/notify)
error[E0432]: unresolved import `inotify`
  --> src/inotify.rs:10:5
   |
10 | use inotify as inotify_sys;
   |     ^^^^^^^^^^^^^^^^^^^^^^ no `inotify` external crate

error[E0432]: unresolved import `mio`
  --> src/inotify.rs:12:5
   |
12 | use mio;
   |     ^^^ no `mio` external crate

error[E0432]: unresolved import `mio_extras`
  --> src/inotify.rs:13:5
   |
13 | use mio_extras;
   |     ^^^^^^^^^^ no `mio_extras` external crate

error[E0432]: unresolved import `mio_extras`
 --> src/error.rs:5:5
  |
5 | use mio_extras;
  |     ^^^^^^^^^^ no `mio_extras` external crate

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
  --> src/inotify.rs:71:32
   |
71 |         if event.mask.contains(EventMask::ISDIR) {
   |                                ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:217:48
    |
217 |                         if event.mask.contains(EventMask::Q_OVERFLOW) {
    |                                                ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:227:48
    |
227 |                         if event.mask.contains(EventMask::MOVED_FROM) {
    |                                                ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:237:52
    |
237 | ...                   if event.mask.contains(EventMask::MOVED_TO) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:260:72
    |
260 | ...                   if event.mask.contains(EventMask::ISDIR) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:273:68
    |
273 | ...                   if event.mask.contains(EventMask::ISDIR) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:284:52
    |
284 | ...                   if event.mask.contains(EventMask::MOVE_SELF) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:295:52
    |
295 | ...                   if event.mask.contains(EventMask::CREATE) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:298:64
    |
298 | ...                   if event.mask.contains(EventMask::ISDIR) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:308:52
    |
308 | ...                   if event.mask.contains(EventMask::DELETE_SELF)
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:309:56
    |
309 | ...                   || event.mask.contains(EventMask::DELETE)
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:313:64
    |
313 | ...                   if event.mask.contains(EventMask::ISDIR) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:323:52
    |
323 | ...                   if event.mask.contains(EventMask::MODIFY) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:331:52
    |
331 | ...                   if event.mask.contains(EventMask::CLOSE_WRITE) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:339:52
    |
339 | ...                   if event.mask.contains(EventMask::CLOSE_NOWRITE) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:347:52
    |
347 | ...                   if event.mask.contains(EventMask::ATTRIB) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `EventMask`
   --> src/inotify.rs:355:52
    |
355 | ...                   if event.mask.contains(EventMask::OPEN) {
    |                                              ^^^^^^^^^ use of undeclared type or module `EventMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:433:29
    |
433 |         let mut watchmask = WatchMask::ATTRIB
    |                             ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:434:15
    |
434 |             | WatchMask::CREATE
    |               ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:435:15
    |
435 |             | WatchMask::DELETE
    |               ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:436:15
    |
436 |             | WatchMask::CLOSE_WRITE
    |               ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:437:15
    |
437 |             | WatchMask::MODIFY
    |               ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:438:15
    |
438 |             | WatchMask::MOVED_FROM
    |               ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:439:15
    |
439 |             | WatchMask::MOVED_TO;
    |               ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:442:30
    |
442 |             watchmask.insert(WatchMask::DELETE_SELF);
    |                              ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:443:30
    |
443 |             watchmask.insert(WatchMask::MOVE_SELF);
    |                              ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:448:30
    |
448 |             watchmask.insert(WatchMask::MASK_ADD);
    |                              ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `WatchMask`
   --> src/inotify.rs:455:38
    |
455 |                     watchmask.remove(WatchMask::MASK_ADD);
    |                                      ^^^^^^^^^ use of undeclared type or module `WatchMask`

error[E0433]: failed to resolve: use of undeclared type or module `Inotify`
   --> src/inotify.rs:520:23
    |
520 |         let inotify = Inotify::init()?;
    |                       ^^^^^^^ use of undeclared type or module `Inotify`

error[E0412]: cannot find type `Inotify` in this scope
  --> src/inotify.rs:38:21
   |
33 | struct EventLoop {
   |                 - help: you might be missing a type parameter: `<Inotify>`
...
38 |     inotify: Option<Inotify>,
   |                     ^^^^^^^ not found in this scope

error[E0412]: cannot find type `WatchDescriptor` in this scope
  --> src/inotify.rs:40:32
   |
33 | struct EventLoop {
   |                 - help: you might be missing a type parameter: `<WatchDescriptor>`
...
40 |     watches: HashMap<PathBuf, (WatchDescriptor, WatchMask, bool)>,
   |                                ^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `WatchMask` in this scope
  --> src/inotify.rs:40:49
   |
33 | struct EventLoop {
   |                 - help: you might be missing a type parameter: `<WatchMask>`
...
40 |     watches: HashMap<PathBuf, (WatchDescriptor, WatchMask, bool)>,
   |                                                 ^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `WatchDescriptor` in this scope
  --> src/inotify.rs:41:20
   |
33 | struct EventLoop {
   |                 - help: you might be missing a type parameter: `<WatchDescriptor>`
...
41 |     paths: HashMap<WatchDescriptor, PathBuf>,
   |                    ^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `WatchDescriptor` in this scope
  --> src/inotify.rs:67:33
   |
64 | fn add_watch_by_event(
   |                      - help: you might be missing a type parameter: `<WatchDescriptor>`
...
67 |     watches: &HashMap<PathBuf, (WatchDescriptor, WatchMask, bool)>,
   |                                 ^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `WatchMask` in this scope
  --> src/inotify.rs:67:50
   |
64 | fn add_watch_by_event(
   |                      - help: you might be missing a type parameter: `<WatchMask>`
...
67 |     watches: &HashMap<PathBuf, (WatchDescriptor, WatchMask, bool)>,
   |                                                  ^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `WatchDescriptor` in this scope
  --> src/inotify.rs:86:33
   |
84 | fn remove_watch_by_event(
   |                         - help: you might be missing a type parameter: `<WatchDescriptor>`
85 |     path: &Option<PathBuf>,
86 |     watches: &HashMap<PathBuf, (WatchDescriptor, WatchMask, bool)>,
   |                                 ^^^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `WatchMask` in this scope
  --> src/inotify.rs:86:50
   |
84 | fn remove_watch_by_event(
   |                         - help: you might be missing a type parameter: `<WatchMask>`
85 |     path: &Option<PathBuf>,
86 |     watches: &HashMap<PathBuf, (WatchDescriptor, WatchMask, bool)>,
   |                                                  ^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `Inotify` in this scope
  --> src/inotify.rs:97:25
   |
97 |     pub fn new(inotify: Inotify, event_fn: Box<dyn EventFn>) -> Result<Self> {
   |                         ^^^^^^^ not found in this scope

error[E0207]: the type parameter `T` is not constrained by the impl trait, self type, or predicates
   --> src/error.rs:150:6
    |
150 | impl<T> From<mio_extras::channel::SendError<T>> for Error {
    |      ^ unconstrained type parameter

error: aborting due to 43 previous errors

Some errors have detailed explanations: E0207, E0412, E0432, E0433.
For more information about an error, try `rustc --explain E0207`.
error: could not compile `notify`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,6 +36,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Cursor position not reported to apps when mouse is moved with button held outside of window
 - No live config update when starting Alacritty with a broken configuration file
 - PTY not drained to the end with the `--hold` flag enabled
+- High CPU usage on BSD with live config reload enabled
 
 ### Removed
 

--- a/alacritty/src/config/monitor.rs
+++ b/alacritty/src/config/monitor.rs
@@ -10,6 +10,11 @@ use alacritty_terminal::thread;
 
 use crate::event::{Event, EventProxy};
 
+#[cfg(any(target_os = "linux", target_os = "macos", target_os = "windows"))]
+const DEBOUNCE_DELAY: Duration = Duration::from_millis(10);
+#[cfg(not(any(target_os = "linux", target_os = "macos", target_os = "windows")))]
+const DEBOUNCE_DELAY: Duration = Duration::from_millis(1000);
+
 pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventProxy) {
     // Canonicalize all paths, filtering out the ones that do not exist.
     paths = paths
@@ -30,7 +35,7 @@ pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventProxy) {
 
     // The Duration argument is a debouncing period.
     let (tx, rx) = mpsc::channel();
-    let mut watcher = match watcher(tx, Duration::from_millis(10)) {
+    let mut watcher = match watcher(tx, DEBOUNCE_DELAY) {
         Ok(watcher) => watcher,
         Err(err) => {
             error!("Unable to watch config file: {}", err);

